crates/holochain/src/conductor/chc/chc_local.rs

```diff
@@ -0,0 +1,385 @@
+                items(&[])
+            );
+            assert_eq!(
+                chc.get_actions_since_hash(Some(hash(9))).await.unwrap(),
+                items(&[0, 1, 2, 3, 4, 5, 6, 7, 8])
+            );
+            assert_eq!(
+                chc.get_actions_since_hash(Some(hash(33))).await.unwrap(),
+                items(&[0, 1, 2, 3, 4, 5, 6, 7, 8])
+            );
+        }
+        .boxed());
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn simple_chc_sync() {
+        use holochain::test_utils::inline_zomes::simple_crud_zome;
+
+        let mut config = ConductorConfig::default();
+        config.chc_namespace = Some("#LOCAL#".to_string());
+        let mut conductor = SweetConductor::from_config(config).await;
+
+        let (dna_file, _, _) = SweetDnaFile::unique_from_inline_zomes(simple_crud_zome())
+            .await
+            .unwrap();
+        let (agent, _) = SweetAgents::alice_and_bob();
+
+        let (cell,) = conductor
+            .setup_app_for_agent("app", agent.clone(), [&dna_file])
+            .await
+            .unwrap()
+            .into_tuple();
+
+        let cell_id = cell.cell_id();
+
+        let top_hash = {
+            let mut dump = conductor
+                .dump_full_cell_state(&cell_id, None)
+                .await
+                .unwrap();
+            assert_eq!(dump.source_chain_dump.records.len(), 3);
+            dump.source_chain_dump.records.pop().unwrap().action_address
+        };
+
+        let new_entry = EntryHashed::from_content_sync(fixt!(Entry));
+        let new_entry_hash = new_entry.as_hash().clone();
+        let create = Create {
+            author: agent.clone(),
+            timestamp: Timestamp::now(),
+            action_seq: 3,
+            prev_action: top_hash,
+            entry_type: fixt!(EntryType),
+            entry_hash: new_entry_hash,
+            weight: EntryRateWeight::default(),
+        };
+        let new_action = ActionHashed::from_content_sync(Action::Create(create));
+        let new_action = SignedActionHashed::sign(&conductor.keystore(), new_action)
+            .await
+            .unwrap();
+
+        {
+            // add some data to the local CHC
+            let m = CHC_LOCAL_MAP.lock();
+            let chc = m.get(&cell_id).unwrap();
+            let actions = chc.get_actions_since_hash(None).await.unwrap();
+            assert_eq!(actions.len(), 3);
+            chc.add_actions(vec![new_action]).await.unwrap();
+            chc.add_entries(vec![new_entry]).await.unwrap();
+        }
+
+        // Check that a sync picks up the new action
+        conductor
+            .inner_handle()
+            .chc_sync(cell_id.clone(), None)
+            .await
+            .unwrap();
+
+        let dump = conductor
+            .dump_full_cell_state(&cell_id, None)
+            .await
+            .unwrap();
+        assert_eq!(dump.source_chain_dump.records.len(), 4);
```

Comment:
you could assert here that the new action's hash matches to be sure that the action was added at the end of the chain